exporter/datadogexporter/config.go

```diff
@@ -54,6 +54,9 @@ type MetricsConfig struct {
 	// metric are kept in memory to calculate deltas
 	DeltaTTL int64 `mapstructure:"delta_ttl"`
 
+	// RemapMetrics ...
+	RemapMetrics bool `mapstructure:"remap_metrics"`
```

Comment:
If this is only intended for internal implementation and not expected to be used by end users (which I think is the case), the common practice is to use a feature gate https://github.com/open-telemetry/opentelemetry-collector/tree/main/featuregate rather a config.